sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/EncryptOptions.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.cryptography;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * A class containing various configuration parameters that can be applied when performing encryption operations.
+ */
+public class EncryptOptions {
+    /**
+     * Initialization vector to be used in the encryption operation using a symmetric algorithm.
+     */
+    @JsonProperty(value = "iv")
+    private final byte[] iv;
+
+    /**
+     * Get additional data to authenticate when performing encryption with an authenticated algorithm.
+     */
+    @JsonProperty(value = "aad")
+    private final byte[] additionalAuthenticatedData;
+
+    /**
+     * Creates an instance of {@link EncryptOptions} with the given parameters.
+     *
+     * @param iv Initialization vector for symmetric algorithms.
+     * @param additionalAuthenticatedData Additional data to authenticate but not encrypt/decrypt when using
+     * authenticated crypto algorithms.
+     */
+    public EncryptOptions(byte[] iv, byte[] additionalAuthenticatedData) {
+        if (iv == null) {
+            this.iv = null;
```

Comment:
See https://apiview.dev/Assemblies/Review/d74052731cb94759804beaa492b7a17a#Azure.Security.KeyVault.Keys.Cryptography.EncryptOptions. Where `iv` is optional (default parameter value of `null`), we should generate an appropriately long one. See https://github.com/Azure/azure-sdk-for-net/blob/275bad601e72a8e308a73aed621e838d1ae20e9b/sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/EncryptOptions.cs#L171-L179 for how/when I did this (I call `Initialize` from local and remote providers and let this class handle if and when it should generate an IV/nonce).

Reply:
...generation of an IV could probably wait closer to GA, but I'd open a bug on that if you punt it for tracking.

Reply:
We do generate a random `iv` of appropriate size depending on the `EncryptionAlgorithm`, [here's an example](https://github.com/vcolin7/azure-sdk-for-java/blob/df0aeb720f67149df6dae0321e7471b6188cbfef/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SymmetricKeyCryptographyClient.java#L77). Although, if the options types are going to also take the algorithm and plaintext/ciphertext we could move that here as well.